src/main/java/com/nike/cerberus/util/AwsIamRoleArnParser.java

```diff
@@ -30,15 +30,43 @@
 
     public static final String AWS_IAM_ROLE_ARN_TEMPLATE = "arn:aws:iam::%s:role/%s";
 
-    public static final String AWS_IAM_PRINCIPAL_ARN_REGEX = "^arn:aws:(iam|sts)::(?<accountId>\\d+?):(role|user|federated-user|assumed-role).*?/(?<roleName>.+)$";
+    /**
```

Comment:
I realized my previous commit earlier today introduced a bug.  Basically, this regex was doing two things:
1. validating ARNs that should be allowed in the DB.
2. the regex was being used for generating role ARNs from other ARN types.

Since we still want to be able to generate role ARNs from instance-profile ARNs (for backwards compatibility) but we don't want them in the DB anymore that means we need two regexes here.